src/bin/scheduler/setup/general.rs

```diff
@@ -273,7 +298,10 @@ fn setup_managed_directories(plans: Vec<Plan>) -> (Vec<Plan>, Vec<SetupFailure>)
     (surviving_plans, failures)
 }
 
-fn clean_up_results_directory(global_config: &GlobalConfig, plans: &[Plan]) -> AnyhowResult<()> {
+fn clean_up_results_directory(
+    global_config: &GlobalConfig,
+    plans: &[Plan],
+) -> Result<(), SetupError> {
     let results_directory_lock = global_config
         .results_directory_locker
         .wait_for_write_lock()?;
```

Comment:
I am surprised that this works as is. Apparently, the conversions implemented above via the `From` trait are applied automatically eg. here, right? Or am I misunderstanding?

Reply:
Yes, the question mark operator is syntatic sugar. Here is a quote:
https://doc.rust-lang.org/reference/expressions/operator-expr.html

> When applied to values of the Result<T, E> type, it propagates errors. If the value is Err(e), then it will return Err(From::from(e)) from the enclosing function or closure. If applied to Ok(x), then it will unwrap the value to evaluate to x.